### ISSUE TITLE
PR 229 - Open/Close input file overhead

### ISSUE THREAD

**Issue by maxbriel** [POSYDON-code/POSYDON-public#230]
In HDF5 read fix #229, the input file is opened and closed every time a new binary is loaded into the BinaryPopulation from a BinaryPopulation file to (re-)evolve, which feels inefficient.

Since we cannot keep the file open the whole time due to issue #224, we could read in a chunk of binaries in one-go before (re-)evolving them.
Is this something we want to implement, is this optimization necessary, and what's the actual overhead of reading one at the time?

Things to consider there is how much space the binaries take up and how the binaries are split between sub-processes?

**Comment by astroJeff** [comment]
I agree this could be optimized. However, it isn't clear to me that this is actually causing a significant time reduction

**Comment by maxbriel** [comment]
It should be pretty bad, having to open the file for each binary you're evolving. However, this should not be a priority.
